lib/rex/post/meterpreter/packet_dispatcher.rb

```diff
@@ -639,6 +638,44 @@ def deregister_inbound_handler(handler)
   attr_accessor :receiver_thread # :nodoc:
   attr_accessor :dispatcher_thread # :nodoc:
   attr_accessor :waiters # :nodoc:
+
+  @tlv_log_file = nil
+
+  def log_packet(packet, packet_type)
+    if framework.datastore['sessiontlvlogging']&.downcase == 'console'
+      log_packet_to_console(packet, packet_type)
+    elsif framework.datastore['sessiontlvlogging']&.downcase =~ /^file:.+/
+      log_packet_to_file(packet, packet_type)
+    end
+  end
+
+  def log_packet_to_console(packet, packet_type)
+    if packet_type == :inbound
+      print "\n%bluRECV%clr: #{packet.inspect}\n"
+    elsif packet_type == :outbound
+      print "\n%redSEND%clr: #{packet.inspect}\n"
+    end
+  end
+
+  def log_packet_to_file(packet, packet_type)
+    path = framework.datastore['sessiontlvlogging'].downcase.split('file:').last
+    pathname = ::Pathname.new(path)
+    if pathname.directory?
+      if @tlv_log_file.nil?
+        @tlv_log_file = ::File.open(::File.dirname(pathname) + '/sessiontlvlogging.txt', 'a+')
+      end
```

Comment:
```suggestion
      @tlv_log_file ||= ::File.open(::File.dirname(pathname) + '/sessiontlvlogging.txt', 'a+')
```

Reply:
Thoughts (that don't need to be implemented): 

Maybe this logic isn't required here, and could be put inside the initial assignment to the `handle_session_tlv_session` 🤔 

If the user deletes this directory/folder path after closing/opening msfconsole, maybe we'll need to recreate the folder structure - with something like:

```
FileUtils.mkdir_p
```

Or maybe we'll just want to log an error, and not guess what the user wants. Just food for thought! 🧠 🥫